### PR TITLE
Installer: --swmh and --sed one-liner options for power users

### PR DESCRIPTION
**Pull request by zijistark**
Completely non-interactive install options.

### CODE REVIEW

File: installer/main.py [hunk 1]
```diff
@@ -881,13 +881,30 @@ def main():
...
+        swmhSelect = False
+        sedSelect = False
+
+        if len(sys.argv) > 1:
+            g_dbgMode = '-D' in sys.argv[1:] or '--debug' in sys.argv[1:]
+            versionMode = '-V' in sys.argv[1:] or '--version' in sys.argv[1:]
+            g_steamMode = '--steam' in sys.argv[1:]
+            g_zijiMode = '-Z' in sys.argv[1:]
+            swmhSelect = '--swmh' in sys.argv[1:]
+            sedSelect = '--sed' in sys.argv[1:]
```
Comment (escalonn) [inline installer/main.py]: btw, slicing never throws indexerrors, and `sys.argv[1:]` always returns either your arguments, or an empty list. Even `[][10:]` returns `[]`. So you don't need this bit to be conditional, and you don’t need the False` init.

**Comment by zijistark** [comment]
Are you sure of that for python2?  Because when I was learning Python for the first time (literally to take over / rewrite the original installer script from @Meneth), I specifically ran into an `IndexError` when I checked `'-D' in sys.argv[1:] when there were no arguments given to the script, thought that was strange (figured Python would just act like a lazy language such as Perl and that the slice would be empty regardless), and fixed the bug by specifically ensuring that that `len(sys.argv) > 1`.

**Comment by zijistark** [comment]
Also, you'll notice that basically everything in `main()` is just a huge mess of explicit variable flags and literal `dict` keys.  It's like that because it's always been, and I've never cared to properly make all those mod "definitions" dynamic (in this case, that'd mean you could select any mod combo from the command-line too). It's kind of embarrassing, but... *shrug*

**Comment by escalonn** [comment]
Odd. i can't reproduce that issue in either python 2 or 3.

**Comment by zijistark** [comment]
OK, thanks for the tip then. I must have been doing something slightly different (maybe just accessing `sys.argv[1]` when `len(sys.argv) == 1`).

**Pull request merged**